multi-node/aws/pkg/cluster/config_test.go

```diff
@@ -0,0 +1,188 @@
+package cluster
+
+import (
+	"bytes"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+const (
+	testVer         = "1"
+	externalDNSName = "test-external-dns-name"
+	keyName         = "test-key-name"
+	region          = "test-region"
+	clusterName     = "test-cluster-name"
+)
+const minimalYamlTemplate = `externalDNSName: %s
+keyName: %s
+region: %s
+clusterName: %s
+`
+
+var minimalConfigYaml string = fmt.Sprintf(
+	minimalYamlTemplate,
+	externalDNSName,
+	keyName,
+	region,
+	clusterName,
+)
+
+func correctMinimalConfig() *Config {
+	config := NewDefaultConfig(testVer)
+	config.ExternalDNSName = externalDNSName
+	config.KeyName = keyName
+	config.Region = region
+	config.ClusterName = clusterName
+	return config
+}
+
+func parseConfig(out *Config, loc string) error {
+	d, err := ioutil.ReadFile(loc)
+	if err != nil {
+		return fmt.Errorf("failed reading config file: %v", err)
+	}
+
+	if err = yaml.Unmarshal(d, &out); err != nil {
+		return fmt.Errorf("failed decoding config file: %v", err)
+	}
+
+	return nil
+}
+
+func TestCorrectMinimalConfig(t *testing.T) {
+	cfg := correctMinimalConfig()
+	if err := cfg.Valid(); err != nil {
+		t.Fatalf("Could not validate: %s\n : %+v\n", err, cfg)
+	}
+}
+
+func writeTestFile(contents string) (string, error) {
+	tpath := filepath.Join(os.TempDir(), fmt.Sprintf("test-file-%d", rand.Int()))
```

Comment:
Unit tests should avoid network/disk IO. We should be able to test the config functionality without hitting disk (or need to restructure code such that the parts we wish to test take a reader/writer interface (not necessarily require a file).